File: SDKs/Aspose.Tasks_Cloud_SDK_for_Ruby/lib/aspose_tasks_cloud/models/outline_code_definition.rb
module AsposeTasksCloud
  # 
  class OutlineCodeDefinition < BaseObject
    attr_accessor :guid, :field_id, :field_name, :_alias, :phonetic_alias, :values, :enterprise, :enterprise_outline_code_alias, :resource_substitution_enabled, :leaf_only, :all_levels_required, :only_table_values_allowed, :masks, :show_indent
    # attribute mapping from ruby-style variable name to JSON key
    def self.attribute_map
      {
        
        # 
        :'guid' => :'Guid',
        
        # 
        :'field_id' => :'FieldId',
        
        # 
        :'field_name' => :'FieldName',
        
        # 
        :'_alias' => :'Alias',
        
        # 
        :'phonetic_alias' => :'PhoneticAlias',
        
        # 
        :'values' => :'Values',
        
        # 
        :'enterprise' => :'Enterprise',
        
        # 
        :'enterprise_outline_code_alias' => :'EnterpriseOutlineCodeAlias',
        
        # 
        :'resource_substitution_enabled' => :'ResourceSubstitutionEnabled',
        
        # 
        :'leaf_only' => :'LeafOnly',
        
        # 
        :'all_levels_required' => :'AllLevelsRequired',
        
        # 
        :'only_table_values_allowed' => :'OnlyTableValuesAllowed',
        
        # 
        :'masks' => :'Masks',
        
        # 
        :'show_indent' => :'ShowIndent'
        
      }
    end

    # attribute type
    def self.swagger_types
      {
        :'guid' => :'String',
        :'field_id' => :'String',
        :'field_name' => :'String',
        :'_alias' => :'String',
        :'phonetic_alias' => :'String',
        :'values' => :'Array<OutlineValue>',
        :'enterprise' => :'BOOLEAN',
        :'enterprise_outline_code_alias' => :'Integer',
        :'resource_substitution_enabled' => :'BOOLEAN',
        :'leaf_only' => :'BOOLEAN',
        :'all_levels_required' => :'BOOLEAN',
        :'only_table_values_allowed' => :'BOOLEAN',
        :'masks' => :'Array<OutlineMask>',
        :'show_indent' => :'BOOLEAN'
        
      }
    end

    def initialize(attributes = {})
      return if !attributes.is_a?(Hash) || attributes.empty?

      # convert string to symbol for hash key
      attributes = attributes.inject({}){|memo,(k,v)| memo[k.to_sym] = v; memo}

      
      if attributes[:'Guid']
        self.guid = attributes[:'Guid']
      end
      
      if attributes[:'FieldId']
        self.field_id = attributes[:'FieldId']
      end
      
      if attributes[:'FieldName']
        self.field_name = attributes[:'FieldName']
      end
      
      if attributes[:'Alias']
        self._alias = attributes[:'Alias']
      end
      
      if attributes[:'PhoneticAlias']
        self.phonetic_alias = attributes[:'PhoneticAlias']
      end
      
      if attributes[:'Values']
        if (value = attributes[:'Values']).is_a?(Array)
          self.values = value
        end
      end
      
      if attributes[:'Enterprise']
        self.enterprise = attributes[:'Enterprise']
      end
      
      if attributes[:'EnterpriseOutlineCodeAlias']
        self.enterprise_outline_code_alias = attributes[:'EnterpriseOutlineCodeAlias']
      end
      
      if attributes[:'ResourceSubstitutionEnabled']
        self.resource_substitution_enabled = attributes[:'ResourceSubstitutionEnabled']
      end
      
      if attributes[:'LeafOnly']
        self.leaf_only = attributes[:'LeafOnly']
      end
      
      if attributes[:'AllLevelsRequired']
        self.all_levels_required = attributes[:'AllLevelsRequired']
      end
      
      if attributes[:'OnlyTableValuesAllowed']
        self.only_table_values_allowed = attributes[:'OnlyTableValuesAllowed']
      end
      
      if attributes[:'Masks']
        if (value = attributes[:'Masks']).is_a?(Array)
          self.masks = value
        end
      end
      
      if attributes[:'ShowIndent']
        self.show_indent = attributes[:'ShowIndent']
      end
      
    end

  end
end

File: SDKs/Aspose.Tasks_Cloud_SDK_for_.NET/src/Com/Aspose/Tasks/Model/OutlineCodeDefinition.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;

namespace Com.Aspose.Tasks.Model {
  public class OutlineCodeDefinition {
    public string Guid { get; set; }

    public string FieldId { get; set; }

    public string FieldName { get; set; }

    public string Alias { get; set; }

    public string PhoneticAlias { get; set; }

    public List<OutlineValue> Values { get; set; }

    public bool? Enterprise { get; set; }

    public int? EnterpriseOutlineCodeAlias { get; set; }

    public bool? ResourceSubstitutionEnabled { get; set; }

    public bool? LeafOnly { get; set; }

    public bool? AllLevelsRequired { get; set; }

    public bool? OnlyTableValuesAllowed { get; set; }

    public List<OutlineMask> Masks { get; set; }

    public bool? ShowIndent { get; set; }

    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class OutlineCodeDefinition {\n");
      sb.Append("  Guid: ").Append(Guid).Append("\n");
      sb.Append("  FieldId: ").Append(FieldId).Append("\n");
      sb.Append("  FieldName: ").Append(FieldName).Append("\n");
      sb.Append("  Alias: ").Append(Alias).Append("\n");
      sb.Append("  PhoneticAlias: ").Append(PhoneticAlias).Append("\n");
      sb.Append("  Values: ").Append(Values).Append("\n");
      sb.Append("  Enterprise: ").Append(Enterprise).Append("\n");
      sb.Append("  EnterpriseOutlineCodeAlias: ").Append(EnterpriseOutlineCodeAlias).Append("\n");
      sb.Append("  ResourceSubstitutionEnabled: ").Append(ResourceSubstitutionEnabled).Append("\n");
      sb.Append("  LeafOnly: ").Append(LeafOnly).Append("\n");
      sb.Append("  AllLevelsRequired: ").Append(AllLevelsRequired).Append("\n");
      sb.Append("  OnlyTableValuesAllowed: ").Append(OnlyTableValuesAllowed).Append("\n");
      sb.Append("  Masks: ").Append(Masks).Append("\n");
      sb.Append("  ShowIndent: ").Append(ShowIndent).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }
  }
  }
